hudi-common/src/main/java/org/apache/hudi/common/model/HoodieRecord.java

```diff
@@ -42,6 +42,8 @@
   public static final String OPERATION_METADATA_FIELD = "_hoodie_operation";
   public static final String HOODIE_IS_DELETED = "_hoodie_is_deleted";
 
+  public static int FILENAME_METADATA_FIELD_POS = 4;
```

Comment:
this could be a separate cleanup task: make constants for all meta fields and adopt them across codebase